test/stripe/file_test.rb

```diff
@@ -62,6 +62,18 @@ class FileTest < Test::Unit::TestCase
         assert file.is_a?(Stripe::File)
       end
 
+      should "be creatable with a URL string" do
+        stub_request(:get, "https://example.com/verification.png")
+          .to_return(status: 200, body: "", headers: {})
+        file = Stripe::File.create(
+          purpose: "dispute_evidence",
+          file: "https://example.com/verification.png",
+          file_link_data: { create: true }
+        )
+        assert_requested :post, "#{Stripe.uploads_base}/v1/files"
+        assert file.is_a?(Stripe::File)
+      end
```

Comment:
This test is a bit of a lie as it passes without the other changes. I didn't see a reasonable way to test this more deeply without introducing a spy or double (which doesn't seem like common practice in the codebase).

This is part of the reason why I pulled out the file parsing into `Util` so I could at least test that.